dds/DCPS/transport/shmem/ShmemTransport.cpp

```diff
@@ -247,22 +250,34 @@ ShmemTransport::ReadTask::ReadTask(ShmemTransport* outer, ACE_sema_t semaphore)
 int
 ShmemTransport::ReadTask::svc()
 {
-  while (true) {
+  while (!stopped_) {
     ACE_OS::sema_wait(&semaphore_);
     if (stopped_) {
       return 0;
     }
```

Comment:
Does this have to stay here if you changed to loop to use `stopped_`?

Reply:
Actually nevermind, we checking it because we're coming out of `sema_wait`, right?

Reply:
Yeah, I didn't want it to do the other stuff after a long wait for the semaphore if it had stopped in between.